### PR TITLE
[AIRFLOW-3753] Replace Flask-OAuthlib with Authlib

### PR DESCRIPTION
**Pull request by lepture**
Make sure you have checked _all_ steps below.

### Jira

- [x] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-3753

### Description

- [x] Here are some details about my PR, including screenshots of any UI changes:

Flask-OAuthlib is deprecated, use Authlib instead. Updated Google and GitHub Enterprise OAuth backends.

### Tests

- [ ] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits

- [x] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [ ] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public functions and the classes in the PR contain docstrings that explain what it does
  - If you implement backwards incompatible changes, please leave a note in the [Updating.md](https://github.com/apache/airflow/blob/master/UPDATING.md) so we can assign it to a appropriate release

### CODE REVIEW

**Review by potiuk — CHANGES_REQUESTED** [review]
I like the change. What worries me a bit is that there are no tests covering the functionality. I don't think there are tests covering the original oauthlib but maybe that's an opportunity to introduce some tests - mocking the behaviour of the oauth ?

File: setup.py [hunk 1]
```diff
@@ -216,9 +216,7 @@ def write_version(filename: str = os.path.join(*["airflow", "git_version"])):
 ]
 grpc = ['grpcio>=1.15.0']
 flask_oauth = [
```
Comment (potiuk) [inline setup.py:218]: I think it should be renamed to "oauth" :)

Comment (lepture) [reply setup.py:218]: Can I change this name? It was named `flask_oauth`.

**Comment by lepture** [comment]
@potiuk It will take me some time to figure out how to write tests for airflow. Maybe in next week.

**Comment by potiuk** [comment]
Hey @lepture - will you have chance to work on it soon :) ?

**Comment by lepture** [comment]
@potiuk I'm afraid that I don't have time right now to figure out how to write the tests.

**Comment by potiuk** [comment]
Ok. I see and understand. Maybe I will try to take a look at that then :).

**Comment by lepture** [comment]
@potiuk if there is an example that would help a lot. I can't find any similar tests.

**Pull request closed without merging**